### PR TITLE
Update GitHub Action deploy steps

### PR DESCRIPTION
**Pull request by natemoo-re**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Update GitHub Pages deployment to use the official [`withastro/action`](https://github.com/withastro/action) GitHub Action

### CODE REVIEW

**Comment by sarah11918** [comment]
Without testing the instructions, everything reads good to me, Nate!

File: src/pages/en/guides/deploy/github.md [hunk 1]
```diff
@@ -11,6 +11,8 @@ You can use [GitHub Pages](https://pages.github.com/) to host an Astro website d
 
 You can deploy an Astro site to GitHub Pages by using [GitHub Actions](https://github.com/features/actions) to automatically build and deploy your site. To do this, your source code must be hosted on GitHub.
 
+Astro maintains the official [`withastro/action`](https://github.com/withastro/action) to deploy your project with very little configuration.
```
Comment (sarah11918) [inline src/pages/en/guides/deploy/github.md:14]:
```suggestion
Astro maintains the official `withastro/action` to deploy your project with very little configuration. Follow the instructions below to deploy your Astro site to GitHub pages, and see [the package README](https://github.com/withastro/action) if you need more information.
```
Not *exactly* sure about this wording, but I'm thinking something that makes it clear that 
a) you don't NEED to click the link (just follow the instructions; you don't need to install anything)
b) you CAN click the link and find a bit more info, if needed

Comment (natemoo-re) [reply src/pages/en/guides/deploy/github.md]: Totally agree!

**Comment by natemoo-re** [comment]
I think your suggestion addressed that concern! I'm walking through the deployment steps right now to make sure everything works as intended!

**Comment by sarah11918** [comment]
When you're happy, @natemoo-re, I'm happy! It's yours to merge.  :)

**Pull request merged**